### ISSUE TITLE
Subsorting to KLabel

### ISSUE THREAD

**Issue by denis-bogdanas** [kframework/k-legacy#261]
I need this feature to better organize labels. This will give shorter rules than by using predicates.
Example code:

syntax XLabel ::= "'X"
syntax KLabel ::= XLabel | "'Y"
rule X:XLabel => 'Y

**Comment by dwightguth** [comment]
This looks like a good feature. It would need to be desugared during
kompile, but otherwise it seems useful to me.

**Comment by grosu** [comment]
I agree, provided that you indeed meant anything of sort XLabel to be rewritten to 'Y, and not

  rule 'X => 'Y

Grigore

**Comment by traiansf** [comment]
does seem reasonable, but keep in mind that KLabel was not a syntactic sort
until now, so desugaring needs some consideration.

best wishes,
Traian

2014-03-03 21:31 GMT+02:00 Grigore Rosu notifications@github.com:

> I agree, provided that you indeed meant anything of sort XLabel to be
> rewritten to 'Y, and not
> 
> rule 'X => 'Y
> 
> Grigore
> 
> ---
> 
> From: dwightguth [notifications@github.com]
> Sent: Monday, March 03, 2014 1:15 PM
> To: kframework/k
> Subject: Re: [k] Subsorting to KLabel (#261)
> 
> This looks like a good feature. It would need to be desugared during
> kompile, but otherwise it seems useful to me.
>

**Comment by dwightguth** [comment]
Well it's still not a syntactic sort. Or it shouldn't be. What we need is a
way to say "these groups of KLabels have a name" and to indicate that
X:XLabel represents an or'd series of predicates, but everything is still
the same sort.

**Comment by traiansf** [comment]
I agree.  But this is approximately the same thing that syntactic sorts do:
 define categories of labels.

2014-03-03 22:00 GMT+02:00 dwightguth notifications@github.com:

> Well it's still not a syntactic sort. Or it shouldn't be. What we need is a
> way to say "these groups of KLabels have a name" and to indicate that
> X:XLabel represents an or'd series of predicates, but everything is still
> the same sort.
> 
> On Mon, Mar 3, 2014 at 11:47 AM, traiansf notifications@github.com
> wrote:
> 
> > does seem reasonable, but keep in mind that KLabel was not a syntactic
> > sort
> > until now, so desugaring needs some consideration.
> > 
> > best wishes,
> > Traian
> > 
> > 2014-03-03 21:31 GMT+02:00 Grigore Rosu notifications@github.com:
> > 
> > > I agree, provided that you indeed meant anything of sort XLabel to be
> > > rewritten to 'Y, and not
> > > 
> > > rule 'X => 'Y
> > > 
> > > Grigore
> > > 
> > > ---
> > > 
> > > From: dwightguth [notifications@github.com]
> > > Sent: Monday, March 03, 2014 1:15 PM
> > > To: kframework/k
> > > Subject: Re: [k] Subsorting to KLabel (#261)
> > > 
> > > This looks like a good feature. It would need to be desugared during
> > > kompile, but otherwise it seems useful to me.
> > > 
> > > On Mon, Mar 3, 2014 at 11:13 AM, Denis Bogdanas <

**Comment by denis-bogdanas** [comment]
This feature don't have a high priority for me. I have a few predicates in the semantics for grouping labels. Subsorting to KLabel would be just a syntactic sugar for the same thing.

**Comment by radumereuta** [comment]
The main problem I see right now with this is with the parser.
I will have to find sorts that are subsorted to KLabel and not subsort them automatically to K, because K and KLabel are disjoint.

I'm assigning this to myself. I will try and solve it when the new parser is working.
